### PR TITLE
Record password reset time

### PR DESCRIPTION
**Pull request by whaught**
Issue https://github.com/google/exposure-notifications-verification-server/issues/575

## Proposed Changes

* Record the time of password change. This will be used to enforce password rotation

**Release Note**

```release-note
Record password reset time
```

### CODE REVIEW

File: pkg/controller/login/select_password.go [hunk 1]
```diff
@@ -31,3 +33,36 @@ func (c *Controller) HandleSelectPassword() http.Handler {
 		c.h.RenderHTML(w, "login/select-password", m)
 	})
 }
+
+func (c *Controller) HandleSubmitNewPassword() http.Handler {
+	logger := c.logger.Named("HandleSubmitNewPassword")
```
Comment (sethvargo) [inline pkg/controller/login/select_password.go:38]:
```suggestion
	logger := c.logger.Named("login.HandleSubmitNewPassword")
```

File: pkg/database/migrations.go [hunk 2]
```diff
@@ -1081,6 +1081,16 @@ func (db *Database) getMigrations(ctx context.Context) *gormigrate.Gormigrate {
 				return nil
 			},
 		},
+		{
+			ID: "00046-AddPasswordLastChangedToUsers",
+			Migrate: func(tx *gorm.DB) error {
+				logger.Debugw("adding email verification required to realm")
+				return tx.Exec("ALTER TABLE users ADD COLUMN IF NOT EXISTS last_password_change DATE DEFAULT CURRENT_DATE").Error
```
Comment (sethvargo) [inline pkg/database/migrations.go:1088]: Does this need to be CURRENT_DATE()? I forget

File: pkg/database/user.go [hunk 3]
```diff
@@ -227,6 +229,21 @@ func (u *User) CreateFirebaseUser(ctx context.Context, fbAuth *auth.Client) (boo
 	return false, nil
 }
 
+// PasswordChanged updates the last password change timestamp of the user.
+func (db *Database) PasswordChanged(email string, t time.Time) error {
+	q := db.db.
+		Model(&User{}).
+		Where("email = ?", email).
+		Update("last_password_change", t.UTC())
```
Comment (sethvargo) [inline pkg/database/user.go:237]: I think we should use UpdateColumn here instead (which will bypass validations and whatnot. Generally faster.

File: pkg/database/migrations.go (hunk 2 above)
Comment (whaught) [reply pkg/database/migrations.go:1088]: I think it's correct as-is https://www.postgresql.org/docs/8.2/functions-datetime.html#FUNCTIONS-DATETIME-CURRENT

**Pull request merged**